### PR TITLE
CP2K: add version 2024.1

### PR DESCRIPTION
**Pull request by RMeli**
Also remove `@master` from DLA-Future and SIRIUS dependencies.

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
Confirmed the version sha256.  Will defer merge to give @mtaillefumier and or @dev-zero time to comment.

**Comment by RMeli** [comment]
Given https://github.com/cp2k/cp2k/issues/3200#issuecomment-1880697749, I think this is finally a good time to switch to CMake as the default build system.

File: var/spack/repos/builtin/packages/cp2k/package.py [hunk 1]
```diff
@@ -439,7 +439,7 @@ def edit(self, spec, prefix):
             "aocc": ["-O2"],
         }
 
-        dflags = ["-DNDEBUG"]
+        dflags = ["-DNDEBUG"] if spec.satisfies("@:2023.2") else []
```
Comment (RMeli) [inline var/spack/repos/builtin/packages/cp2k/package.py:442]: https://github.com/cp2k/cp2k/pull/3172

**Review by tldahlgren — APPROVED** [review]

**Review by mtaillefumier — APPROVED** [review]
agreed. cmake should be the default the current release.

**Pull request merged**